Source/WebCore/Modules/indexeddb/IDBKeyRangeData.h

```diff
@@ -32,64 +32,61 @@ namespace WebCore {
 
 class IDBKey;
 
-struct IDBKeyRangeData {
-    IDBKeyRangeData()
-        : isNull(true)
-    {
-    }
-
-    static IDBKeyRangeData allKeys()
-    {
-        IDBKeyRangeData result;
-        result.isNull = false;
-        result.lowerKey = IDBKeyData::minimum();
-        result.upperKey = IDBKeyData::maximum();
-        return result;
-    }
-
+class IDBKeyRangeData {
+public:
+    struct Data {
+        IDBKeyData lowerKey;
+        IDBKeyData upperKey;
+        
+        bool lowerOpen { false };
+        bool upperOpen { false };
+        
+        Data isolatedCopy() const { return { lowerKey.isolatedCopy(), upperKey.isolatedCopy(), lowerOpen, upperOpen }; }
+    };
+    using NullOrData = std::variant<std::nullptr_t, Data>;
```

Comment:
Do we really need `std::nullptr_t` to represent null? Can we just do:
```
bool isNull() {
    return lowerKey.isNull() && upperKey.isNull();
}
bool isValid() {
    return lowerKey.isInvalid() || upperKey.isInvalid();
}
```

Reply:
That is a change in behavior that fails many tests